src/common/locallaplacian.c

```diff
@@ -135,83 +157,40 @@ static inline void gauss_reduce_sse2(
   // blur, store only coarse res
   const int cw = (wd-1)/2+1, ch = (ht-1)/2+1;
 
-  // this version is inspired by opencv's pyrDown_ :
-  // - allocate 5 rows of ring buffer (aligned)
-  // - for coarse res y
-  //   - fill 5 coarse-res row buffers with 1 4 6 4 1 weights (reuse some from last time)
-  //   - do vertical convolution via sse and write to coarse output buf
-
-  const int stride = ((cw+8)&~7); // assure sse alignment of rows
-  float *ringbuf = dt_alloc_align(64, sizeof(*ringbuf)*stride*5);
-  float *rows[5] = {0};
-  int rowj = 0; // we initialised this many rows so far
-
-  for(int j=1;j<ch-1;j++)
-  {
-    // horizontal pass, convolve with 1 4 6 4 1 kernel and decimate
-    for(;rowj<=2*j+2;rowj++)
-    {
-      float *const row = ringbuf + (rowj % 5)*stride;
-      const float *const in = input + rowj*wd;
 #ifdef _OPENMP
 #pragma omp parallel for default(none) \
-      dt_omp_firstprivate(cw, in, row) \
+      dt_omp_firstprivate(cw, ch, input)   \
```

Comment:
to compile with recent GCC you need to add `wd` and `coarse` in the firstprivate list.